crates/gpui/src/platform/windows/dispatcher.rs

```diff
@@ -0,0 +1,108 @@
+use std::thread::JoinHandle;
+
+use async_task::Runnable;
+use flume::Sender;
+use parking::Parker;
+use parking_lot::Mutex;
+use windows::Win32::System::Threading::GetCurrentThreadId;
+
+use crate::{PlatformDispatcher, TaskLabel};
+
+pub(crate) struct WindowsDispatcher {
+    background_sender: Sender<(Runnable, Option<TaskLabel>)>,
+    main_sender: Sender<Runnable>,
+    background_threads: Vec<JoinHandle<()>>,
+    parker: Mutex<Parker>,
+}
+
+impl WindowsDispatcher {
+    pub(crate) fn new(main_sender: Sender<Runnable>) -> Self {
+        let parker = Mutex::new(Parker::new());
+        let (background_sender, background_receiver) =
+            flume::unbounded::<(Runnable, Option<TaskLabel>)>();
+        let background_threads = (0..std::thread::available_parallelism()
+            .map(|i| i.get())
+            .unwrap_or(1))
+            .map(|_| {
+                let receiver = background_receiver.clone();
+                std::thread::spawn(move || {
+                    for (runnable, label) in receiver {
+                        if let Some(label) = label {
+                            log::debug!("TaskLabel: {label:?}");
+                        }
+                        runnable.run();
+                    }
+                })
+            })
+            .collect::<Vec<_>>();
+        Self {
+            background_sender,
+            main_sender,
+            background_threads,
+            parker,
+        }
+    }
+}
+
+extern "C" {
+    static MAIN_THREAD_ID: u32;
+}
```

Comment:
This is unnecessary, you can just query the `GetCurrentThreadId` in Dispatcher::new() and assume it's main_thread_id.

Better to remove this & and it's related link to cpp

Reply:
I did this because I wasn't sure if the dispatcher initialization would be done in the main thread, but if that's ok, I'll do it.

Reply:
> wasn't sure if the initialization would be done in the main thread

It is, linux platform assumes the same.

Reply:
I did it